packages/ui/src/components/Designer/RightSidebar/DetailView/index.tsx

```diff
@@ -35,6 +36,7 @@ const DetailView = (
   useEffect(() => {
     const newWidgets: typeof widgets = {
       AlignWidget: (p) => <AlignWidget {...p} {...props} options={options} />,
+      TextFormatWidget: (p) => <TextFormatWidget {...p} {...props} options={options} />,
```

Comment:
I can see what you have done here and why, but I think this needs to be more generic, and not tied to an implementation just for the text schema.

I think we could have a `BooleanButtonsWidget` which could be generic enough to pass in the names of attributes and SVG details to be rendered.

Reply:
Using a quick test I could make this:

![Screenshot 2024-07-16 at 06 32 12](https://github.com/user-attachments/assets/01fe2b87-cd39-46d2-a253-02f8ee004a80)

I think we could do a similar buttons widget for text alignment too. I will try this later today.

Reply:
<img width="394" alt="Screenshot 2024-07-16 at 4 18 34 PM" src="https://github.com/user-attachments/assets/718243c6-032e-49e2-a85a-0a994659f74d">

@peteward How about like this?

And we can configure buttons like below

```ts
export enum Formatter {
  STRIKETHROUGH = 'strikethrough',
  UNDERLINE = 'underline',
  ALIGNMENT = 'alignment',
  VERTICAL_ALIGNMENT = 'verticalAlignment',
}


{
    title: i18n('schemas.text.format'),
    widget: 'ButtonGroup',
    buttons: [
      { key: Formatter.STRIKETHROUGH, svg: TextStrikethroughIcon, type: 'boolean' },
      { key: Formatter.UNDERLINE, svg: TextUnderlineIcon, type: 'boolean' },
      { key: Formatter.ALIGNMENT, svg: TextAlignLeftIcon, type: 'select', value: DEFAULT_ALIGNMENT },
      { key: Formatter.ALIGNMENT, svg: TextAlignCenterIcon, type: 'select', value: ALIGN_CENTER},
      { key: Formatter.ALIGNMENT, svg: TextAlignRightIcon, type: 'select', value: ALIGN_RIGHT},
      { key: Formatter.VERTICAL_ALIGNMENT, svg: TextVerticalAlignTopIcon, type: 'select', value: DEFAULT_VERTICAL_ALIGNMENT },
      { key: Formatter.VERTICAL_ALIGNMENT, svg: TextVerticalAlignMiddleIcon, type: 'select', value: VERTICAL_ALIGN_MIDDLE},
      { key: Formatter.VERTICAL_ALIGNMENT, svg: TextVerticalAlignBottomIcon, type: 'select', value: VERTICAL_ALIGN_BOTTOM},
    ]
    span: 16,
 };
```

Reply:
I will push the code, so that you can check. And please feel free to give the feedback.

Reply:
this is excellent, I really like it!

I had just played around with something similar but splitting the buttons into groups (but it was hard to fit one one line)

![Screenshot 2024-07-16 at 10 51 09](https://github.com/user-attachments/assets/295e0835-29c3-4b8c-b795-c64d7883dce4)


Your code is better though and would give the flexibility to split them if desired 👍 

I think one final improvement would be to use SVGs that are more familiar to users working with text, like:

![Screenshot 2024-07-16 at 10 54 38](https://github.com/user-attachments/assets/95eb5114-deae-42b1-9b92-3b4a13383adb)
![Screenshot 2024-07-16 at 10 55 31](https://github.com/user-attachments/assets/909b7f2c-3fa2-45c0-8151-56025b970fe1)

We could always add these later though.

@hand-dot, what do you think of this PR?

Reply:
> I think one final improvement would be to use SVGs that are more familiar to users working with text.

Yeah agreed, but I was just following the same icon from upper align icons.

Reply:
I understand.

Personally, I think it would look clearer to have different icons from the widget alignment ones to distinguish the functionality.

Reply:
<img width="430" alt="Screenshot 2024-07-16 at 8 19 01 PM" src="https://github.com/user-attachments/assets/d7e6c0d2-ea4b-40e4-8589-54451446b3ba">

Updated.

Reply:
great work!

Reply:
yes yes! Really nice.